### ISSUE TITLE
Assign PP3 using PROVEAN (Protein Variation Effect Analyzer) 

### ISSUE THREAD

**Issue by kennethto** [genewhisper/genewhisper#8]
We need to add PROVEAN in order to assign PP3 (http://provean.jcvi.org/index.php). Someone please look into this and determine how to add wrap this into oncotator  
We need to run all the variants via PROVEAN determine the output

**Comment by mzhakun** [comment]
I don't think we have resources to add this tool to oncotator, but I'm 'sure we can figure out a way how to use it as a separate tool which is works with oncotator. and the wrap it together as one application. However, we have to determine what would be input and desired output.

**Comment by mzhakun** [comment]
I spend some time studying PROVEAN tool and i have some issues with it. 
First of all Provean web server tool which takes data in format  
<chromosome>,<position>,<reference allele>,<variant allele>,<comment(optional)> 
see more info here http://provean.jcvi.org/help.php#genome_input_format
and tool which is they provide for download is little different tools and difference is in input they required. Tool for download required this type of input:
Required arguments:

> 	 -q <string>, --query <string>
> 	   Query protein sequence filename in fasta format
> 	 -v <string>, --variation <string>
> 	   Variation filename containing a list of variations:
> 	     one entry per line in HGVS notation,
> 	     e.g.: G105C, F508del, Q49dup, Q49_P50insC, Q49_R52delinsLI

I don't understand what is **list of variation** and why does it require to input protein sequense instead of Parameters related to Genome Variation (<chromosome>,<position>,<reference allele>,<variant allele>,<comment(optional)>). 

We can work around protein sequence by referring to UniProt using Hugo_Symbol in Oncotator and that way we will get Protein sequence FASTA file. But we still gonna need that  list of variation file which is in .var format and looking like this: 
```

P72R
G105C
K370del
H178_H179insPHP
L22_W23delinsQS
```
Ken advised me to use p. number but without p. , I've tried it but I've got an error.  Looks like it has to be more than 1 or 2 numbers and they have to be different that p. in order to make that prediction. Heres The error with p. number 

```
mzhakun@mzhakun-VirtualBox:/usr/local/share/data/provean/examples$ provean.sh -q P04637.fasta -v P04637.var
## PROVEAN v1.1 output ##
# Query sequence file:	P04637.fasta
# Variation file:	P04637.var
# Protein database:	/home/mzhakun/blastdb/nr
Invalid position: 809 (length of query sequence: 256)
Invalid position: 816 (length of query sequence: 256)
No variations entered

```
Ayushi and Anwar, 
Can you guys install Provean and figure out what is **list of variation** file and where can we get it from. Also, run some tests and let me know what you came up with. 

Thanks.

**Comment by seaspray09** [comment]
The downloadable package of PROVEAN only contains the PROVEAN Protein function, and not the  PROVEAN Protein Batch or Genome Variants tools.  The Genome Variants tools of PROVEAN that we are looking to use are only available on the web server.

**Comment by seaspray09** [comment]
For the list of variants, the missense variants that are annotated in UniProt are in the humsaver.txt file at the link below:

http://www.uniprot.org/docs/humsavar

Under the column "AA change" column, it shows amino acid changes in the P dot format.

**Comment by mzhakun** [comment]
@aagraw16, that's what I mentioned in my statement above online tool and tool for download are different, but we still can work around it. By using protein sequence in FASTA format. 
Whatever  if you installed the software and run sample you should be aware that file of variation .var is having a format like this:
```
P72R
G105C
K370del
H178_H179insPHP
L22_W23delinsQS 
```
which is not format in AA change column. 
If you get Provean working with any protein sequense in FASTA format file and p. number in AA column please share a screenshot.

**Comment by seaspray09** [comment]
We could try to convert the format in "AA change column".  The AA change column is using 3-letter abbreviations of amino acids, and the .var file is using the 1-letter abbreviation (see list below)

For example, "P72R" corresponds to "Pro72Arg"

Ala | A | Alanine
Arg | R | Arginine
Asn | N | Asparagine
Asp | D | Aspartic acid
Cys | C | Cysteine
Gln | Q | Glutamine
Glu | E | Glutamic acid
Gly | G | Glycine
His | H | Histidine
Ile | I | Isoleucine
Leu | L | Leucine
Lys | K | Lysine
Met | M | Methionine
Phe | F | Phenylalanine
Pro | P | Proline
Ser | S | Serine
Thr | T | Threonine
Trp | W | Tryptophan
Tyr | Y | Tyrosine
Val | V | Valine

Asx | B | Asn or Asp
Glx | Z | Gln or Glu
Xle | J | Leu or Ile
Sec | U | Selenocysteine (UGA)
Pyl | O | Pyrrolysine (UAG)
Unk | X | Unknown

**Comment by mzhakun** [comment]
@seaspray09 .var file requires at least 2 modifications. And what is the rule to change it to the format you mentioned?

**Comment by seaspray09** [comment]
So basically each amino acid name would have to be converted to the single letter abbreviation per the table above, hopefully that makes sense.  There isn't a particular rule to change it to this format, it would just make it easier to use the data from the "AA change column"  in PROVEAN.

@kennethto 
However, I am not sure where to find the insertions and deletions in the databases.  For example, I am not able to find the K370del (deletion of lysine at position 370) in relation to having some effect on TP53 (even in the literature searches on Pubmed), but this should be described somewhere.  I would think there would be some description, even if is non-deleterious.

The other thing with PROVEAN is whether it accounts for all the variations that could be determined as "deleterious".  The tool was validated with substitutions, insertions, and deletions, but not for frameshift and nonsense mutations.  So any deleterious effect coming from a frameshift mutation will not be accounted for in PROVEAN.

**Comment by mzhakun** [comment]
@seaspray09  have you run any test using that data in AA column?

**Comment by seaspray09** [comment]
No i have not been able to yet, but the .var file (at least for the missense variations) should be derived from there right?  Or is it not even reading the following:

P72R
G105C
 
I am not understanding why it is not even recognizing these as variations when you ran it; is it creating some new line characters that are not supposed to be there in the .var text file?

**Comment by kennethto** [comment]
@seaspray09  K370del  is in http://p53.iarc.fr/TP53GeneVariations.aspx. You can just search for K370del in Protein description list. And it looks like it is in this paper (https://www.ncbi.nlm.nih.gov/pubmed/8001119) but I don't have access to read it.

**Comment by seaspray09** [comment]
For the Provean functionality we are probably going to have to come up with our own tool or find something else because it does not give us the information we need and would be too much of a hassle to work with when things are put on AWS.

**Comment by kennethto** [comment]
use filter to search as following
1) search the geneId
2)search the c. with that given gene
3)search the latest date (lastEvaled)
4)output the clinical significant
